src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -0,0 +1,1841 @@
+ 				}else {
+ 					System.out.println("INFO: Route with default non-timing-driven clock router");
+ 	 				GlobalSignalRouting.defaultClkRoute(clk, this.design.getDevice());
+ 				}
+ 			}
+			this.preserveNet(clk);
+ 		}
+	}
+	
+	/**
+	 * Adds and initialize a regular signal net to the list of routing targets.
+	 * @param net The net to be added for routing.
+	 * @param multiSLR A flag to indicate if the device has multiple SLRs, for the sake of avoiding unnecessary check for single-SLR devices.
+	 */
+	protected void addNetConnectionToRoutingTargets(Net net, boolean multiSLR) {
+		net.unroute();
+		this.numWireNetsToRoute++;;
+		this.createsNetWrapperAndConnections(net, this.config.getBoundingBoxExtensionX(), this.config.getBoundingBoxExtensionY(),multiSLR);
+	}
+	
+	/**
+	 * Adds a static net to the static net routing target list.
+	 * @param staticNet The static net in question, i.e. VCC or GND.
+	 */
+	protected void addStaticNetRoutingTargets(Net staticNet){
+		List<SitePinInst> sinks = new ArrayList<>();
+		for(SitePinInst sink : staticNet.getPins()){
+			if(sink.isOutPin()) continue;
+			sinks.add(sink);
+		}
+		
+		if(sinks.size() > 0 ) {
+			for(SitePinInst sink : sinks) {
+				this.addReservedNode(sink.getConnectedNode(), staticNet);
+			}
+			this.staticNetAndRoutingTargets.put(staticNet, sinks);
+		}else {
+			this.preserveNet(staticNet);
+			this.numNotNeedingRoutingNets++;	
+		}
+	}
+	
+	protected void addStaticNetRoutingTargets(Net staticNet, List<SitePinInst> sinks) {
+		this.staticNetAndRoutingTargets.put(staticNet, sinks);
+	}
+	
+	/**
+	 * Routes static nets with preserved resources list supplied to avoid conflicting nodes.
+	 * @param preservedConnectedNodesToPins The list of preserved nodes for other nets.
+	 */
+	private void routeStaticNets(List<Node> preservedConnectedNodesToPins){
+		GlobalSignalRouting.setDesignRoutethruHelper(this.design, this.routethruHelper);
+		
+		for(Net net : this.staticNetAndRoutingTargets.keySet()){
+			for(SitePinInst sink : this.staticNetAndRoutingTargets.get(net)) {
+				this.preservedNodes.remove(sink.getConnectedNode());
+			}
+		}
+		
+		RouterHelper.invertPossibleGndPinsToVccPins(this.design, this.design.getGndNet());
+		
+		// If connections of other nets are routed first, used resources should be preserved.
+		Set<Node> unavailableNodes = getAllUsedNodesOfRoutedConnections();
+		unavailableNodes.addAll(this.preservedNodes.keySet());
+		// When the connections of other nets are not routed yet, 
+		// the nodes connected to pins of other nets should be supplied and preserved.
+		if(preservedConnectedNodesToPins != null) unavailableNodes.addAll(preservedConnectedNodesToPins);
+		unavailableNodes.addAll(this.rnodesCreated.keySet());
+		
+		for(Net net : this.staticNetAndRoutingTargets.keySet()){
+			System.out.println("INFO: Route " + net.getSinkPins().size() + " pins of " + net);
+			Map<SitePinInst, List<Node>> spiRoutedNodes = GlobalSignalRouting.routeStaticNet(net, unavailableNodes);
```

Comment:
Looks like the map value of `staticNetAndRoutingTargets` is not used here?

Reply:
They are not used here. They are used in printing design info and checking the number of static net pins that should be routed. It helps identifying problems if any.